### DISCUSSION TITLE
query is giving %ge calculation as 0 for all columns.  when however all should be over 1%

### DISCUSSION THREAD

**Discussion by jgrover21** [dbeaver/dbeaver#21288] [Q&A] [answered]
### Description

Following query is not giving %ge.  it is giving 0 for all %ge calculations

select count(*) as non_pas_ids,
	(count(*) filter (where c.age  is null)) /count(*) * 100 as age_missing
,(count(*) filter (where c.sex  is null))/count(*) * 100 as sex_missing
	,(count(*) filter (where c.bmi  is null))/count(*) * 100 as bmi_missing
	,(count(*) filter (where c.race  is null)) as race_missing
	,(count(*) filter (where c.ethnic  is null)) as ethnic_missing
from (
select distinct on (a.subject)
    a.subject,
	a.age,
	a.sex,
	a.bmi,
	a.race,
	a.ethnic,
	b.flag,
	b.subject
from
	cvad.enrollment a
left join cvad.pas_id_list b
on
	a.subject = b.subject) c
where flag is not null

### DBeaver Version

23.1.1

### Operating System

windows

### Database and driver

postgresql Azure

**Comment by E1izabeth** [comment, marked as answer]
Please try rewriting your condition this way 
`(count(*) filter (where c.age is null)) * 100 /count(*)`
or this way
`((count(*) filter (where c.age is null))::float/count(*)::float) *100`

You get 0 because when database divide integer to integer, the result is integer, but you expect to get floating point number.
